taxi/taxibench_pandas_ibis.py

```diff
@@ -272,20 +274,63 @@ def etl_ibis(
     omnisci_server_worker.create_database(database_name, delete_if_exists=delete_old_database)
 
     if create_new_table:
-        # TODO t_import_pandas, t_import_ibis = omnisci_server_worker.import_data_by_ibis
-        t0 = time.time()
-        omnisci_server_worker.import_data(
-            table_name=table_name,
-            data_files_names=data_files_names,
-            files_limit=files_limit,
-            columns_names=columns_names,
-            columns_types=columns_types,
-            header=False,
-        )
-        etl_times["t_readcsv"] = time.time() - t0
-        # etl_times["t_readcsv"] = t_import_pandas + t_import_ibis
+        etl_times["t_readcsv"] = 0
+        schema_table = ibis.Schema(names=columns_names, types=columns_types)
+        if import_mode == "copy-from":
+            # Create table and import data for ETL queries
+            omnisci_server_worker.create_table(
+                table_name=table_name, schema=schema_table, database=database_name,
+            )
+            table_import = omnisci_server_worker.database(database_name).table(table_name)
+
+            for file_to_import in data_files_names[:files_limit]:
+                t0 = timer()
+                table_import.read_csv(file_to_import, header=False, quotechar="", delimiter=",")
+                etl_times["t_readcsv"] += round((timer() - t0) * 1000)
+
+        elif import_mode == "pandas":
+            # Datafiles import
+            t_import_pandas, t_import_ibis = omnisci_server_worker.import_data_by_ibis(
+                table_name=table_name,
+                data_files_names=data_files_names,
+                files_limit=files_limit,
+                columns_names=columns_names,
+                columns_types=columns_types,
+                header=0,
+                nrows=None,
+                compression_type="gzip" if filename.endswith("gz") else None,
+                validation=validation,
+            )
+
+            etl_times["t_readcsv"] = round((t_import_pandas + t_import_ibis) * 1000)
+
+        elif import_mode == "fsi":  # Currently work for single file
+            unzip_name = None
+            if data_files_names[0].endswith("gz"):
+                import gzip
+                import os
+
+                unzip_name = "/tmp/taxibench-fsi.csv"
+
+            for file_to_import in data_files_names[:files_limit]:
+                try:
+                    if file_to_import.endswith("gz"):
```

Comment:
```suggestion
                    if file_to_import.endswith(".gz"):
```
otherwise you risk getting to a e.g. `.tgz` which _cannot_ be uncompressed like this

Reply:
This for taxi dataset only which is in that form. Of course we can check the form and report user if something is unexpected